text/0171-cloudfront-redesign.md

```diff
@@ -0,0 +1,477 @@
+use your own domain name, such as `www.example.com`, you can add an alternate domain name to your distribution.
+
+```ts
+new Distribution(this, 'myDist', {
+  origin: Origin.fromBucket(myBucket),
+  aliases: ['www.example.com']
+})
+```
+
+CloudFront distributions use a default certificate (`*.cloudfront.net`) to support HTTPS by default. If you want to support HTTPS with your own domain
+name, you must associate a certificate with your distribution that contains your domain name. The certificate must be present in the AWS Certificate
+Manager (ACM) service in the US East (N. Virginia) region; the certificate may either be created by ACM, or created elsewhere and imported into ACM.
+
+```ts
+const myCertificate = new certmgr.DnsValidatedCertificate(this, 'mySiteCert', {
+  domainName: 'www.example.com',
+  hostedZone,
+});
+new Distribution(this, 'myDist', {
+  origin: Origin.fromBucket(myBucket),
+  certificate: myCertificate,
+});
+```
+
+Note that in the above example the aliases are inferred from the certificate and do not need to be explicitly provided.
+
+## Caching Behaviors
+
+Each distribution has a default cache behavior which applies to all requests to that distribution; additional cache behaviors may be specified for a
+given URL path pattern. Cache behaviors allowing routing with multiple origins, controlling which HTTP methods to support, whether to require users to
+use HTTPS, and what query strings or cookies to forward to your origin, among other behaviors.
+
+The properties of the default cache behavior can be adjusted as part of the distribution creation. The following example shows configuring the HTTP
+methods and viewer protocol policy of the cache.
+
+```ts
+const myWebDistribution = new Distribution(this, 'myDist', {
+  origin: Origin.fromHTTPServer({
+    domainName: 'www.example.com',
+    originProtocolPolicy: OriginProtocolPolicy.HTTPS_ONLY
+  }),
+  behavior: {
+    allowedMethods: AllowedMethods.ALL,
+    viewerProtocolPolicy: ViewerProtocolPolicy.REDIRECT_TO_HTTPS
+  }
+});
+```
+
+Additional cache behaviors can be specified at creation, or added to the origin(s) after the initial creation. These additional cache behaviors enable
+customization for a specific set of resources based on a URL path pattern. For example, we can add a behavior to `myWebDistribution` to override the
+default time-to-live (TTL) for all of the images.
+
+```ts
+myWebDistribution.origin.addBehavior('/images/*.jpg', {
+  viewerProtocolPolicy: ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
+  defaultTtl: cdk.Duration.days(7),
+});
+```
+
+## Multiple Origins
+
+A distribution may have multiple origins in addition to the default origin; each additional origin must have (at least) one behavior to route requests
+to that origin. A common pattern might be to serve all static assets from an S3 bucket, but all dynamic content served from a web server. The
```

Comment:
> A distribution may have multiple origins in addition to the default origin; each additional origin must have (at least) one behavior to route requests to that origin.

I find this a little non-intuitive. Above, `behavior` was a separate thing, that you passed when creating `Distribution` next to origin:

```ts
new Distribution(this, 'Distr', {
  origin: ...
  behavior: {
    ...
  },
});
````

But below, you have `additionalOrigins` which don't specify any behaviors... So is this example correct, or is it still missing something?